backend/substrapp/tasks/tasks_compute_task.py

```diff
@@ -251,7 +256,10 @@ def _run(
         # start build_image timer
         timer.start()
 
-        wait_for_image_built(ctx.function, channel_name)
+        image_builder.wait_for_image_built(ctx.function, channel_name)
```

Comment:
I _think_  we only want to wait for the image to be built if we are the owner of the function, but I might be wrong 🤔